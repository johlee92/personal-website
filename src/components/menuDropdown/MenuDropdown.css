.menu-dropdown {
    background-color: darkgray;
}

ul.menu-dropdown {
    list-style: none;
    margin: 0;
    padding: 0;
    text-align: center;
}

li.menu-dropdown {
    padding: 5px 0;
}

li.menu-dropdown:hover {
    background-color: black;
    color: white;
    font-weight: bold;
}