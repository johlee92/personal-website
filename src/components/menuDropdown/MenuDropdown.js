import React, { Component } from 'react';
import './MenuDropdown.css';
import { Link } from 'react-router-dom';

class MenuDropdown extends Component {
    constructor(props){
        super(props);
    }

    //requires the parent to pass in a function to close the menu when its been handled

    render() {
        return (
            <div className="menu-dropdown header-mobile">
                <ul className="menu-dropdown">
                    <li className="menu-dropdown" onClick={this.props.menuDisplayChange}><Link to='/' style={{ textDecoration: 'none' }}>Home</Link></li>
                    <li className="menu-dropdown" onClick={this.props.menuDisplayChange}><Link to='/about' style={{ textDecoration: 'none' }}>About</Link></li>
                    <li className="menu-dropdown" onClick={this.props.menuDisplayChange}><Link to='/journal' style={{ textDecoration: 'none' }}>Journal</Link></li>
                    <li className="menu-dropdown" onClick={this.props.menuDisplayChange}><Link to='/projects' style={{ textDecoration: 'none' }}>Projects</Link></li>
                </ul>
            </div> 
        )
    }
}

export default MenuDropdown;