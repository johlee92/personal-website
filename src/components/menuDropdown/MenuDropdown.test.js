import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import MenuDropdown from './MenuDropdown';
import { BrowserRouter } from 'react-router-dom';

describe(`MenuDropdown component`, () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <MenuDropdown />
            </BrowserRouter>
            , div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the component as expected', () => {
        const tree = renderer
            .create(
                <BrowserRouter>
                    <MenuDropdown />
                </BrowserRouter>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();  
    });
})
