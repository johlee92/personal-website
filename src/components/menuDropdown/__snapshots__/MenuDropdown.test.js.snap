// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuDropdown component renders the component as expected 1`] = `
<div
  className="menu-dropdown header-mobile"
>
  <ul
    className="menu-dropdown"
  >
    <li
      className="menu-dropdown"
    >
      <a
        href="/"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        Home
      </a>
    </li>
    <li
      className="menu-dropdown"
    >
      <a
        href="/about"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        About
      </a>
    </li>
    <li
      className="menu-dropdown"
    >
      <a
        href="/journal"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        Journal
      </a>
    </li>
    <li
      className="menu-dropdown"
    >
      <a
        href="/projects"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        Projects
      </a>
    </li>
  </ul>
</div>
`;
